docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,259 @@
+
+```toml
+[MessageQueue]
+Protocol = 'tcp'
+Host = 'localhost'
+Port = 1883
+Type = 'mqtt'
+PublishTopic = 'edgex/events/<device-profile-name>/<device-name>'
+SubscribeEnabled = true
+SubscibeTopic = 'edgex/events/#'
+[MessageQueue.Optional]
+    # Default MQTT Specific options that need to be here to enable evnironment variable overrides of them
+    # Client Identifiers
+    ClientId ="edgex-core-data"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "messagebus"
+```
+
+#### Application Services
+
+##### [MessageBus]
+
+Similar to above, the Application Services `MessageBus` configuration will change to allow for secure connection to the Message Bus. The `Username` & `Password` have been replaced with `Authmode` & `SecretPath` for secure connections. See **[Secure Connections](#secure-connections)** section below for details.
+
+```toml
+[MessageBus.Optional]
+    # MQTT Specific options
+    # Client Identifiers
+    ClientId ="<app sevice key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "messagebus"
+```
+
+##### [Binding]
+
+The `Binding` configuration section will require change for the subscribe topics scheme describe in the [Message Bus Topics](#message-bus-topics) section above to filter for Events from specific device profiles or devices. `SubscribeTopic` will change from a string property containing a single topic to the `SubscribeTopics` string property containing a comma separated list of topics. This allows for the flexibility for the property to be a single topic with the `#` wild card so the Application Service receives all Events as it does today.
+
+Receive only Events from the `Random-Integer-Device` and `Random-Boolean-Device` profiles
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/Random-Integer-Device, edgex/events/Random-Boolean-Device"
```

Comment:
Do we really need "edgex" as part of the topic schema?

Reply:
An app service may be using multiple pub/subs. I like having `edgex` in the topic as it explicitly describes the source. It is all configuration, so folks can tweak as they see fit.